cmd/noms/noms_sync.go

```diff
@@ -77,10 +77,21 @@ func runSync(args []string) int {
 		lastProgressCh <- last
 	}()
 
+	sourceRef := types.NewRef(sourceObj)
+	sinkRef, _ := sinkDataset.MaybeHeadRef()
+	nonFF := false
 	err = d.Try(func() {
 		defer profile.MaybeStartProfile().Stop()
+		sinkDataset.Pull(sourceStore, sourceRef, p, progressCh)
+
 		var err error
-		sinkDataset, err = sinkDataset.Pull(sourceStore, types.NewRef(sourceObj), p, progressCh)
+		if sourceRef.Height() > sinkRef.Height() {
```

Comment:
Actually, we do have some tests for noms sync that use our ClientTestSuite
thing. Maybe I can leverage that

On Fri, Aug 5, 2016 at 10:31 AM Rafael Weinstein <notifications@github.com>
wrote:

> In cmd/noms/noms_sync.go
> <https://github.com/attic-labs/noms/pull/2271#discussion_r73729522>:
>
> >  		var err error
> > -		sinkDataset, err = sinkDataset.Pull(sourceStore, types.NewRef(sourceObj), p, progressCh)
> > +		if sourceRef.Height() > sinkRef.Height() {
>
> ok
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/attic-labs/noms/pull/2271/files/9840f4594b4ac912e30b79cf185d00ff363c196e#r73729522>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMnImuXCFRWHg2VJHg26bTKdPOGPsXeYks5qc3NVgaJpZM4Jc_2B>
> .
>